aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncDeltaSync.java

```diff
@@ -0,0 +1,540 @@
+        this.awsAppSyncClient = awsAppSyncClient;
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    //Initialize the Delta Sync persistence mechanism
+    private void initializeIfRequired() {
+        synchronized(initializationLock) {
+            Log.d(TAG, "In initialize method");
+            if (awsAppSyncDeltaSyncSqlHelper == null ) {
+                //Setup up the local store
+                Log.d(TAG, "Initializing the database");
+                awsAppSyncDeltaSyncSqlHelper = new AWSAppSyncDeltaSyncSqlHelper(mContext);
+            }
+
+            if( dbHelper == null ) {
+                dbHelper = new AWSAppSyncDeltaSyncDBOperations(awsAppSyncDeltaSyncSqlHelper);
+            }
+
+            if (!recordCreatedOrFound) {
+
+                AWSAppSyncDeltaSyncDBOperations.DeltaSyncRecord record;
+                if ((record = dbHelper.getRecordByKey(getKey())) == null ) {
+                    this.id = dbHelper.createRecord(getKey(), lastRunTimeInMilliSeconds);
+                }
+                else {
+                    this.id = record.id;
+                    this.lastRunTimeInMilliSeconds = record.lastRunTimeInMilliSeconds;
+                }
+                deltaSyncObjects.put(id,this);
+                recordCreatedOrFound = true;
+            }
+
+        }
+    }
+
+    void setBaseQuery(Query baseQuery) {
+        this.baseQuery = baseQuery;
+    }
+
+    void setBaseQueryCallback(GraphQLCall.Callback<Query.Data> callback) {
+        this.baseQueryCallback = callback;
+    }
+
+    void setSubscription(Subscription subscription) {
+        this.subscription = subscription;
+    }
+
+    void setSubscriptionCallback( AppSyncSubscriptionCall.Callback subscriptionCallback) {
+        this.subscriptionCallback = subscriptionCallback;
+    }
+
+
+    void setDeltaQuery(Query deltaQuery) {
+        this.deltaQuery = deltaQuery;
+    }
+
+    void setDeltaQueryCallback(GraphQLCall.Callback<Query.Data> callback) {
+        this.deltaQueryCallback = callback;
+    }
+
+    void setBaseRefreshIntervalInSeconds(long baseRefreshIntervalInSeconds) {
+        this.baseRefreshIntervalInSeconds = baseRefreshIntervalInSeconds;
+    }
+
+    private String getKey( ) {
+        return  "" + baseQuery + subscription + deltaQuery;
+    }
+
+    Long execute(final boolean forceFetch) {
+
+        //Initialize the Delta Sync Machinery if required.
+        initializeIfRequired();
+
+        this.deltaSyncOperationFailed = false;
+        new Thread( new Runnable() {
+            @Override
+            public void run() {
+                final CountDownLatch baseQueryCountdownLatch = new CountDownLatch(1);
+                Log.v(TAG, "Delta Sync: Starting Sync process");
+
+                //Run base query from the cache
+                runBaseQuery(AppSyncResponseFetchers.CACHE_ONLY);
+                if (deltaSyncOperationFailed ) {
+                    return;
```

Comment:
In case of a failure, we should start a retry timer right? Is that done in some other part of the code?

Reply:
retry added.